cpu/arm7_common/include/arm_cpu.h

```diff
@@ -25,6 +25,16 @@ void cpu_clock_scale(uint32_t source, uint32_t target, uint32_t *prescale);
 
 void arm_reset(void);
 
+/**
+ * @brief Interrupt stack canary value
+ *
+ * @note 0xe7fe is the ARM Thumb machine code equivalent of asm("bl #-2\n") or
+ * 'while (1);', i.e. an infinite loop.
+ * @internal
+ */
+#define STACK_CANARY_WORD   (0xE7FEE7FEu)
```

Comment:
Currently thumb mode is not used on ARMv7. (Parts of the inline assembly cannot be mapped to thumb instructions.)

Reply:
Do you know a good tool? [hex2arm](http://armconverter.com/hextoarm/) produces 
```asm
B	#0
B	#2
```
from fee7fee7.

Entering that into [arm2hex](http://armconverter.com/) yields

```
E7FE
E7FF
```
for Thumb

and
```
EAFFFFFE
EAFFFFFE
```

for ARM mode. So I guess I'll use just that.

Reply:
(I would have expected to have to change `bl #-2` to `bl #-4` for ARM since instructions are 32 instead of 16 bit)

Reply:
I used

```C
void foo() {
	while (1) {}
}
```

ran `arm-none-eabi-gcc -Os -c -o foo.o foo.c` and then `arm-none-eabi-objdump -d foo.o` which yielded:

```
test.o:     file format elf32-littlearm


Disassembly of section .text:

00000000 <foo>:
   0:	eafffffe 	b	0 <foo>
```

So `0xeafffffe` should be correct